File: app/src/main/java/info/nightscout/androidaps/plugins/PumpInsightLocal/app_layer/history/history_events/StartOfTBREvent.java
package info.nightscout.androidaps.plugins.PumpInsightLocal.app_layer.history.history_events;

import info.nightscout.androidaps.plugins.PumpInsightLocal.utils.ByteBuf;

public class StartOfTBREvent extends HistoryEvent {

    private int amount;
    private int duration;

    @Override
    public void parse(ByteBuf byteBuf) {
        amount = byteBuf.readUInt16LE();
        duration = byteBuf.readUInt16LE();
    }

    public int getAmount() {
        return amount;
    }

    public int getDuration() {
        return duration;
    }
}

File: pump/insight/src/main/kotlin/app/aaps/pump/insight/app_layer/history/history_events/StartOfTBREvent.kt
package app.aaps.pump.insight.app_layer.history.history_events

import app.aaps.pump.insight.utils.ByteBuf

class StartOfTBREvent : HistoryEvent() {

    internal var amount = 0
        private set
    internal var duration = 0
        private set

    override fun parse(byteBuf: ByteBuf) {
        byteBuf.let {
            amount = it.readUInt16LE()
            duration = it.readUInt16LE()
        }
    }
}